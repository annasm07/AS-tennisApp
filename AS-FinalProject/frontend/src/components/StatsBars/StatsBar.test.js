import {render} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import StatsBar from './StatsBars';

describe('When invoked a StatsBar component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    user: {user: {playerId: '123cde'}},
  });
  let title;
  let p1Value;
  let p2Value;
  beforeEach(() => {
    title = 'Double Faults';
    p1Value = 10;
    p2Value = 8;
  });
  test('Should render StatsBar component', () => {
    const statsBar = render(
      <Provider store={store}>
        <StatsBar title={title} p1Value={p1Value} p2Value={p2Value} />
      </Provider>,
    );
    expect(statsBar).toMatchSnapshot();
  });
});
