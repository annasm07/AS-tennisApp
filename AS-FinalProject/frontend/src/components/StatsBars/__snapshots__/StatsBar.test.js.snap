// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a StatsBar component Should render StatsBar component 1`] = `
<View
  style={
    Object {
      "marginVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "width": "90%",
      }
    }
  >
    <Text>
      10
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Double Faults
    </Text>
    <Text>
      8
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#E5E5E5",
        "borderRadius": 5,
        "display": "flex",
        "flexDirection": "row",
        "height": 14,
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#F5BF00",
              "borderBottomLeftRadius": 5,
              "borderRightColor": "#000",
              "borderRightWidth": 1,
              "borderTopLeftRadius": 5,
              "height": 14,
              "paddingHorizontal": 20,
              "width": "50%",
            },
            Object {
              "width": "56%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#F7A65B",
              "borderBottomRightRadius": 5,
              "borderLeftColor": "#000",
              "borderLeftWidth": 1,
              "borderTopRightRadius": 5,
              "height": 14,
              "paddingHorizontal": 20,
              "width": "50%",
            },
            Object {
              "width": "44%",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
